packages/slate-editor/src/extensions/galleries/components/Gallery/GalleryTile.tsx

```diff
@@ -1,37 +1,93 @@
 import classNames from 'classnames';
 import type { CSSProperties } from 'react';
-import React from 'react';
+import React, { forwardRef } from 'react';
 
-import { ImageSizeWarning, ImageWithLoadingPlaceholder } from '#components';
+import { Button, ImageSizeWarning, ImageWithLoadingPlaceholder } from '#components';
+import { Crop, Delete } from '#icons';
 
 import styles from './GalleryTile.module.scss';
 
-interface Props {
+export interface Props {
+    caption?: string;
     className?: string;
-    url: string;
-    imageWidth: number;
+    clone?: boolean;
+    dragging?: boolean;
     imageHeight: number;
+    imageWidth: number;
+    insertPosition?: 'before' | 'after';
+    isInteractive?: boolean;
+    onCaptionChange: (caption: string) => void;
+    onCrop: () => void;
+    onDelete: () => void;
+    padding?: GalleryPadding;
     style?: CSSProperties;
+    url: string;
     withBorderRadius: boolean;
     withSizeWarning?: boolean;
+    withSmallButtons?: boolean;
 }
 
-export function GalleryTile({
-    className,
-    url,
-    imageWidth,
-    imageHeight,
-    style = {},
-    withBorderRadius,
-    withSizeWarning = false,
-}: Props) {
+export const GalleryTile = forwardRef<HTMLDivElement, Props>(function GalleryTile(
+    {
+        caption,
+        className,
+        clone = false,
+        dragging,
+        imageHeight,
+        imageWidth,
+        insertPosition,
+        isInteractive,
+        onCaptionChange,
+        onCrop,
+        onDelete,
+        padding,
+        style = {},
+        url,
+        withBorderRadius,
+        withSizeWarning = false,
+        withSmallButtons = false,
+        ...props
+    },
+    ref,
+) {
     return (
         <div
             className={classNames(styles.GalleryTile, className, {
                 [styles.withBorderRadius]: withBorderRadius,
+                [styles.interactive]: isInteractive,
+                [styles.dragging]: dragging,
+                [styles.insertBefore]: insertPosition === 'before',
+                [styles.insertAfter]: insertPosition === 'after',
+                [styles.withMediumPadding]: padding === GalleryPadding.M,
+                [styles.withLargePadding]: padding === GalleryPadding.L,
             })}
+            ref={ref}
             style={style}
         >
+            {clone && <div className={styles.Clone} />}
+            {!dragging && (
+                <div className={styles.Overlay}>
+                    <div className={styles.Actions}>
+                        <CropButton isSmall={withSmallButtons} onCrop={onCrop} />
+                        <DeleteButton isSmall={withSmallButtons} onDelete={onDelete} />
+                    </div>
+                    <div className={styles.DragHandle} {...props} tabIndex={-1} />
+                    <div
+                        className={classNames(styles.Caption, {
+                            [styles.visible]: caption !== '',
+                        })}
+                    >
```

Comment:
The problem here with the outline is that on full width blocks, the outline is rendered as "inner" instead of the regular one.

I've tried removing it and keeping the regular outline and that fixes it. Obviously, there's now the problem that the outline overflows the container a little bit but I think it's fine in this case as otherwise I don't see a way to fix this as the inner outline takes away some space from the block.

cc @alpower 

![Screenshot 2024-02-26 at 12 34 51](https://github.com/prezly/slate/assets/4209081/3a086726-92b0-48f3-af96-497fee55c756)